GitUI/CommandsDialogs/Menus/WorkingDirectoryToolStripSplitButton.cs

```diff
@@ -0,0 +1,302 @@
+        /// </summary>
+        protected RepositoryHistoryUIService RepositoryHistoryUIService
+            => _use_property_instead_repositoryHistoryUIService ?? throw new InvalidOperationException("The button is not initialized");
+
+        /// <summary>
+        ///  Gets the current instance of the UI commands.
+        /// </summary>
+        protected GitUICommands UICommands
+            => (_use_property_instead_getUICommands ?? throw new InvalidOperationException("The button is not initialized")).Invoke();
+
+        /// <summary>
+        ///  Initializes the menu item.
+        /// </summary>
+        /// <param name="getUICommands">The method that returns the current instance of UI commands.</param>
+        public void Initialize(Func<GitUICommands> getUICommands, RepositoryHistoryUIService repositoryHistoryUIService,
+                               StartToolStripMenuItem startToolStripMenuItem, ToolStripMenuItem closeToolStripMenuItem)
+        {
+            Translator.Translate(this, AppSettings.CurrentTranslation);
+
+            _use_property_instead_getUICommands = getUICommands;
+            _use_property_instead_repositoryHistoryUIService = repositoryHistoryUIService;
+            _startToolStripMenuItem = startToolStripMenuItem;
+            _closeToolStripMenuItem = closeToolStripMenuItem;
+
+            // Initilize toolstip menu items
+            // ----------------------------------------
+
+            _tsmiCategorisedRepos = new(_startToolStripMenuItem.FavouriteRepositoriesMenuItem.Text, _startToolStripMenuItem.FavouriteRepositoriesMenuItem.Image)
+            {
+                Tag = _excludeFromFilterMarker
+            };
+
+            _tsmiOpenLocalRepository = new(_startToolStripMenuItem.OpenRepositoryMenuItem.Text, _startToolStripMenuItem.OpenRepositoryMenuItem.Image)
+            {
+                ShortcutKeys = _startToolStripMenuItem.OpenRepositoryMenuItem.ShortcutKeys,
+                Tag = _excludeFromFilterMarker
+            };
+            _tsmiOpenLocalRepository.Click += (s, e) => _startToolStripMenuItem.OpenRepositoryMenuItem.PerformClick();
+
+            _tsmiCloseRepo = new(_closeToolStripMenuItem.Text, _closeToolStripMenuItem.Image)
+            {
+                Tag = _excludeFromFilterMarker
+            };
+            _tsmiCloseRepo.Click += (hs, he) => _closeToolStripMenuItem.PerformClick();
+
+            _tsmiRecentReposSettings = new(_configureWorkingDirMenu.Text)
+            {
+                Tag = _excludeFromFilterMarker
+            };
+            _tsmiRecentReposSettings.Click += (hs, he) =>
+            {
+                using (FormRecentReposSettings frm = new())
+                {
+                    frm.ShowDialog(OwnerForm);
+                }
+
+                RefreshContent();
+            };
+        }
+
+        protected override void OnButtonClick(EventArgs e)
+        {
+            base.OnButtonClick(e);
+            ShowDropDown();
+        }
```

Comment:
This is too different from a `ButtonClick` handler.
As a result the dropdown cannot be closed by clicking the button again.

Reply:
No, it's the same action, though it's behaving slightly different from the event handler. Good catch!